comet/broker.py

```diff
@@ -528,17 +466,21 @@ def saveDataset(hash, ds, root):
 
 async def tree(root):
     """Return a list of all nodes in the given tree."""
-    tree = dict()
-    async with lock_datasets:
-        for n in datasets_of_root[root]:
-            tree[n] = datasets[n]
+    async with await lock_manager.lock("datasets_lock"):
+        datasets_of_root = json.loads(await redis.execute("hget", "datasets_of_root", root))
```

Comment:
This looks like it's probably scriptable.

Also, can datasets_of_root not be a native redis set (or even sorted set)?